README.md

```diff
@@ -108,43 +108,86 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Hazelcast serializes all your objects into an instance of `Data`. `Data` is the binary representation of an object. When Hazelcast serializes an object into `Data`, it first checks whether the object is an instance of `IdentifiedDataSerializable`, if not it checks if it is an instance of `Portable` and serializes it accordingly. For the following types Hazelcast optimizes the serialization a user can not override this behaviour.
+- A `boolean` type which is a primitive on Node.js can be serialized as a `Boolean` for Java server-side.
+- A character which is a string with length one can be serialized as a `Character` for Java server-side.
+- A `number` type which is a primitive on Node.js can be serialized as `Byte`, `Short`, `Integer`, `Float`, `Double` for Java server-side.
+- A `Long` type which represents a 64 bit two's-complement integer value can be serialized as a `Long` for Java server-side.
+- A `string` type which is a primitive on Node.js can be serialized as a `String` for Java server-side.
+- Arrays of the above types can be serialized as `boolean[]`, `char[]`, `byte[]`, `short[]`, `int[]`, `float[]`, `double[]`, `long[]`, `string[]` for Java server-side respectively. 
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Note that if the object is not instance of any explicit type, Node.js client uses `JSON Serialization`. The default behaviour can be changed by using a `Custom Serialization`.
```

Comment:
Note that if the object is not an instance of any explicit type, Node.js client uses `JSON Serialization` by default. This default behavior can be changed by using a `Custom Serialization`.

(Please give a link to Custom Serialization section here by making it a cross-reference in the above sentence).

Reply:
Is `explicit type` a defined thing? If I have to define the types that Hazelcast serialization takes care of, I would say primitive types. Yet primitive types also do not cover it because we also take care of serializing `Date` objects which are not primitive.

Reply:
This comment shows collapsed because the line has changed but my concern is not resolved.